apps/frontpage/components/docs/footer/form.tsx

```diff
@@ -0,0 +1,145 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
```

Comment:
Note the lack of a `use "client"` directive.

TL;DR: That directive is only needed at the server/client boundary. Including it here, which is always rendered inside Footer and therefore always client, caused TS errors.

See [this comment](https://github.com/vercel/next.js/discussions/46795#discussioncomment-5248407) and [this reply](https://github.com/vercel/next.js/discussions/46795#discussioncomment-5937396) for details.